lib/archethic/db/embedded_impl/chain_index.ex

```diff
@@ -371,14 +371,36 @@ defmodule Archethic.DB.EmbeddedImpl.ChainIndex do
 
     filename = chain_addresses_path(db_path, genesis_address)
 
-    case :ets.lookup(:archethic_db_last_index, genesis_address) do
-      [{_, ^new_address, _}] ->
-        :ok
+    write_last_chain_transaction? =
+      case :ets.lookup(:archethic_db_last_index, genesis_address) do
+        [{_, ^new_address, _}] -> false
+        [{_, _, chain_unix_time}] when unix_time < chain_unix_time -> false
+        _ -> true
+      end
 
-      _ ->
-        :ok = File.write!(filename, encoded_data, [:binary, :append])
-        true = :ets.insert(:archethic_db_last_index, {genesis_address, new_address, unix_time})
-        :ok
+    write_last_chain_transaction(
```

Comment:
I think using the if statement would simplify the api of the `write_last_chain_transaction/6` fn

Reply:
no problem I will rewrite it